src/main/java/com/yelp/nrtsearch/server/luceneserver/highlights/FastVectorHighlighter.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2022 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver.highlights;
+
+import com.yelp.nrtsearch.server.luceneserver.field.TextBaseFieldDef;
+import java.io.IOException;
+import java.util.Map;
+import org.apache.lucene.document.FieldType;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.highlight.DefaultEncoder;
+import org.apache.lucene.search.vectorhighlight.BaseFragmentsBuilder;
+import org.apache.lucene.search.vectorhighlight.FieldQuery;
+import org.apache.lucene.search.vectorhighlight.FragListBuilder;
+import org.apache.lucene.search.vectorhighlight.ScoreOrderFragmentsBuilder;
+import org.apache.lucene.search.vectorhighlight.SimpleFragListBuilder;
+import org.apache.lucene.search.vectorhighlight.SimpleFragmentsBuilder;
+import org.apache.lucene.search.vectorhighlight.SingleFragListBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The FVH highlighter class for a search query. The highlighted field must be indexed with term
+ * vector with positions and offsets. And the multivalue fields are always fragmented and scored
+ * discretely.
+ */
+public class FastVectorHighlighter implements Highlighter {
```

Comment:
I think it might be good to rename this class like FastVectorHighlighterProvider or some name with NRT as prefix to avoid having the same name as org.apache.lucene.search.vectorhighlight.FastVectorHighlighter. But it is totally your call since Elasticsearch is actually using the same name in their repo 😀

Reply:
Agree. We can add NRT prefix to all builtin highlighters. but the `customHighlighterName`

Reply:
plainHighlighter actually has the same issue. Lucene calls it Highlighter, and ES add prefix"plain" to distinguish them